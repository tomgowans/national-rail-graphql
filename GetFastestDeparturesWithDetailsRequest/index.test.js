const GetResponse = require("./index");

it("sends a reques with minimal parameters correctly", () => {
  expect.assertions(1);

  return expect(
    GetResponse(
      {
        crs: "ECR"
      },
      "TOKEN_VALUE"
    )
  ).resolves.toMatchSnapshot();
});

it("sends a reques with more parameters correctly", () => {
  expect.assertions(1);

  return expect(
    GetResponse(
      {
        numRows: "12",
        crs: "ECR",
        filterCrs: ["STP"]
      },
      "TOKEN_VALUE"
    )
  ).resolves.toMatchSnapshot();
});
