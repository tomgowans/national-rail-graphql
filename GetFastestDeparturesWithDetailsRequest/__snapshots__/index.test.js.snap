// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sends a reques with minimal parameters correctly 1`] = `
Object {
  "$": Object {
    "xmlns:lt": "http://thalesgroup.com/RTTI/2012-01-13/ldb/types",
    "xmlns:lt2": "http://thalesgroup.com/RTTI/2014-02-20/ldb/types",
    "xmlns:lt3": "http://thalesgroup.com/RTTI/2015-05-14/ldb/types",
    "xmlns:lt4": "http://thalesgroup.com/RTTI/2015-11-27/ldb/types",
    "xmlns:lt5": "http://thalesgroup.com/RTTI/2016-02-16/ldb/types",
    "xmlns:lt6": "http://thalesgroup.com/RTTI/2017-02-02/ldb/types",
    "xmlns:lt7": "http://thalesgroup.com/RTTI/2017-10-01/ldb/types",
  },
  "crs": "ECR",
  "departures": Object {
    "destination": Object {
      "$": Object {
        "crs": "STP",
      },
      "service": Array [
        Object {
          "cancelReason": "This train has been cancelled because of a fault with the signalling system earlier today",
          "delayReason": "This train has been delayed by a fault with the signalling system earlier today",
          "destination": Object {
            "location": Array [
              Object {
                "crs": "PBO",
                "locationName": "Peterborough",
              },
            ],
          },
          "eta": "17:27",
          "etd": "17:27",
          "length": "12",
          "operator": "Thameslink",
          "operatorCode": "TL",
          "origin": Object {
            "location": Array [
              Object {
                "crs": "HRH",
                "locationName": "Horsham",
              },
            ],
          },
          "platform": "1",
          "serviceID": "2tI4/cosU7DYIHRO7z0DLg==",
          "serviceType": "train",
          "sta": "17:15",
          "std": "17:15",
          "subsequentCallingPoints": Object {
            "callingPointList": Object {
              "callingPoint": Array [
                Object {
                  "crs": "LBG",
                  "et": "17:41",
                  "length": "12",
                  "locationName": "London Bridge",
                  "st": "17:30",
                },
                Object {
                  "crs": "BFR",
                  "et": "17:47",
                  "length": "12",
                  "locationName": "London Blackfriars",
                  "st": "17:36",
                },
                Object {
                  "crs": "CTK",
                  "et": "17:49",
                  "length": "12",
                  "locationName": "City Thameslink",
                  "st": "17:38",
                },
                Object {
                  "crs": "ZFD",
                  "et": "17:51",
                  "length": "12",
                  "locationName": "Farringdon Underground",
                  "st": "17:41",
                },
                Object {
                  "crs": "STP",
                  "et": "17:55",
                  "length": "12",
                  "locationName": "London St Pancras (Intl)",
                  "st": "17:45",
                },
                Object {
                  "crs": "FPK",
                  "et": "18:02",
                  "length": "12",
                  "locationName": "Finsbury Park",
                  "st": "17:52",
                },
                Object {
                  "crs": "SVG",
                  "et": "18:22",
                  "length": "12",
                  "locationName": "Stevenage",
                  "st": "18:13",
                },
                Object {
                  "crs": "HIT",
                  "et": "18:27",
                  "length": "12",
                  "locationName": "Hitchin",
                  "st": "18:19",
                },
                Object {
                  "crs": "ARL",
                  "et": "18:33",
                  "length": "12",
                  "locationName": "Arlesey",
                  "st": "18:25",
                },
                Object {
                  "crs": "BIW",
                  "et": "18:38",
                  "length": "12",
                  "locationName": "Biggleswade",
                  "st": "18:30",
                },
                Object {
                  "crs": "SDY",
                  "et": "18:41",
                  "length": "12",
                  "locationName": "Sandy",
                  "st": "18:34",
                },
                Object {
                  "crs": "SNO",
                  "et": "18:48",
                  "length": "12",
                  "locationName": "St Neots",
                  "st": "18:41",
                },
                Object {
                  "crs": "HUN",
                  "et": "18:55",
                  "length": "12",
                  "locationName": "Huntingdon",
                  "st": "18:48",
                },
                Object {
                  "crs": "PBO",
                  "et": "19:10",
                  "length": "12",
                  "locationName": "Peterborough",
                  "st": "19:05",
                },
              ],
            },
          },
        },
      ],
    },
  },
  "generatedAt": "2019-12-17T17:27:28.1271088+00:00",
  "locationName": "East Croydon",
  "nrccMessages": Object {
    "message": Array [
      "Major disruption between London and Stevenage. More details can be found in <A href=\\"http://nationalrail.co.uk/service_disruptions/238310.aspx\\">Latest Travel News.</A>",
      "A passenger has been taken ill on a train at St Albans City and as a result some lins are blocked.  More information can be found in<A href=\\"http://nationalrail.co.uk/service_disruptions/238335.aspx\\"> Latest Travel News.</A>",
      "Disruption through Redhill. More information can be found in <A href=\\"http://nationalrail.co.uk/service_disruptions/238336.aspx\\">Latest Travel News.</A>",
    ],
  },
  "platformAvailable": "true",
}
`;

exports[`sends a reques with more parameters correctly 1`] = `
Object {
  "$": Object {
    "xmlns:lt": "http://thalesgroup.com/RTTI/2012-01-13/ldb/types",
    "xmlns:lt2": "http://thalesgroup.com/RTTI/2014-02-20/ldb/types",
    "xmlns:lt3": "http://thalesgroup.com/RTTI/2015-05-14/ldb/types",
    "xmlns:lt4": "http://thalesgroup.com/RTTI/2015-11-27/ldb/types",
    "xmlns:lt5": "http://thalesgroup.com/RTTI/2016-02-16/ldb/types",
    "xmlns:lt6": "http://thalesgroup.com/RTTI/2017-02-02/ldb/types",
    "xmlns:lt7": "http://thalesgroup.com/RTTI/2017-10-01/ldb/types",
  },
  "crs": "ECR",
  "departures": Object {
    "destination": Object {
      "$": Object {
        "crs": "STP",
      },
      "service": Array [
        Object {
          "cancelReason": "This train has been cancelled because of a fault with the signalling system earlier today",
          "delayReason": "This train has been delayed by a fault with the signalling system earlier today",
          "destination": Object {
            "location": Array [
              Object {
                "crs": "PBO",
                "locationName": "Peterborough",
              },
            ],
          },
          "eta": "17:27",
          "etd": "17:27",
          "length": "12",
          "operator": "Thameslink",
          "operatorCode": "TL",
          "origin": Object {
            "location": Array [
              Object {
                "crs": "HRH",
                "locationName": "Horsham",
              },
            ],
          },
          "platform": "1",
          "serviceID": "2tI4/cosU7DYIHRO7z0DLg==",
          "serviceType": "train",
          "sta": "17:15",
          "std": "17:15",
          "subsequentCallingPoints": Object {
            "callingPointList": Object {
              "callingPoint": Array [
                Object {
                  "crs": "LBG",
                  "et": "17:41",
                  "length": "12",
                  "locationName": "London Bridge",
                  "st": "17:30",
                },
                Object {
                  "crs": "BFR",
                  "et": "17:47",
                  "length": "12",
                  "locationName": "London Blackfriars",
                  "st": "17:36",
                },
                Object {
                  "crs": "CTK",
                  "et": "17:49",
                  "length": "12",
                  "locationName": "City Thameslink",
                  "st": "17:38",
                },
                Object {
                  "crs": "ZFD",
                  "et": "17:51",
                  "length": "12",
                  "locationName": "Farringdon Underground",
                  "st": "17:41",
                },
                Object {
                  "crs": "STP",
                  "et": "17:55",
                  "length": "12",
                  "locationName": "London St Pancras (Intl)",
                  "st": "17:45",
                },
                Object {
                  "crs": "FPK",
                  "et": "18:02",
                  "length": "12",
                  "locationName": "Finsbury Park",
                  "st": "17:52",
                },
                Object {
                  "crs": "SVG",
                  "et": "18:22",
                  "length": "12",
                  "locationName": "Stevenage",
                  "st": "18:13",
                },
                Object {
                  "crs": "HIT",
                  "et": "18:27",
                  "length": "12",
                  "locationName": "Hitchin",
                  "st": "18:19",
                },
                Object {
                  "crs": "ARL",
                  "et": "18:33",
                  "length": "12",
                  "locationName": "Arlesey",
                  "st": "18:25",
                },
                Object {
                  "crs": "BIW",
                  "et": "18:38",
                  "length": "12",
                  "locationName": "Biggleswade",
                  "st": "18:30",
                },
                Object {
                  "crs": "SDY",
                  "et": "18:41",
                  "length": "12",
                  "locationName": "Sandy",
                  "st": "18:34",
                },
                Object {
                  "crs": "SNO",
                  "et": "18:48",
                  "length": "12",
                  "locationName": "St Neots",
                  "st": "18:41",
                },
                Object {
                  "crs": "HUN",
                  "et": "18:55",
                  "length": "12",
                  "locationName": "Huntingdon",
                  "st": "18:48",
                },
                Object {
                  "crs": "PBO",
                  "et": "19:10",
                  "length": "12",
                  "locationName": "Peterborough",
                  "st": "19:05",
                },
              ],
            },
          },
        },
      ],
    },
  },
  "generatedAt": "2019-12-17T17:27:28.1271088+00:00",
  "locationName": "East Croydon",
  "nrccMessages": Object {
    "message": Array [
      "Major disruption between London and Stevenage. More details can be found in <A href=\\"http://nationalrail.co.uk/service_disruptions/238310.aspx\\">Latest Travel News.</A>",
      "A passenger has been taken ill on a train at St Albans City and as a result some lins are blocked.  More information can be found in<A href=\\"http://nationalrail.co.uk/service_disruptions/238335.aspx\\"> Latest Travel News.</A>",
      "Disruption through Redhill. More information can be found in <A href=\\"http://nationalrail.co.uk/service_disruptions/238336.aspx\\">Latest Travel News.</A>",
    ],
  },
  "platformAvailable": "true",
}
`;
